benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp

```diff
@@ -0,0 +1,525 @@
+        archive( cereal::make_nvp("vector" , _vec_f) );
+    }
+    
+    void serialize__Vector_to__JSON_file(const std::string & filename_string){
+        std::ofstream os(filename_string);
+        {
+            cereal::JSONOutputArchive archive_out(os);
+            serialize(archive_out);
+        }
+        return;
+    }
+
+    std::string serialize__Vector_to__JSON(){
+        std::stringstream os;
+        {
+            cereal::JSONOutputArchive archive_out(os);
+            serialize(archive_out);
+        }
+        return os.str();
+    }
+
+    // json_string provided must have expected format as created by cereal::JSONOutputArchive.
+    std::vector<float> deserialize__JSON_to__Vector(const std::string & json_string){
+        std::stringstream is (json_string);
+        {
+            cereal::JSONInputArchive archive_in(is);
+            serialize(archive_in);
+        }
+        return _vec_f;
+    }
+    
+    std::vector<float> deserialize__JSON_file_to__Vector(const std::string & filename_string){
+        std::ifstream is(filename_string);
+        {
+            cereal::JSONInputArchive archive_in(is);
+            serialize(archive_in);
+        }
+        return _vec_f;
+    }
+
+    void load_random(int i_vec_size){
+        for(int i = 0; i < i_vec_size; i++) 
+        {
+            float f_rand_value = get_random_float_using_whole_range();
+            _vec_f.push_back(f_rand_value);
+        }
+    }
+
+    size_t get_size(){
+        return _vec_f.size();
+    }
+
+    std::vector<float> get_vec(){
+        return _vec_f;
+    }
+
+    void clear_vector(){
+        _vec_f.clear();
+    }
+    
+    // Load vector randomly with desired size.
+    myVectorFloats(int i_size_vec) {
+        load_random(i_size_vec);
+    };
+
+    myVectorFloats() = default;
+};
+
+
+
+
+
+
+
+
+
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// benchmarks: Deserialize memory JSON stream to memory structures.
+//
+// This test benchmarks spent time deserializing a memory JSON formatted stream
+// of std::vector<float>, to a memory std::vector<float> with as many floats as
+// in JSON vector representation loaded.
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+static void Cereal__Retrive_float_vector(benchmark::State &state)
+{
+    try{
+        // Create a json_string related to test conditions.
+        int i_vec_size = state.range(0);
+        std::string json_string;
+        size_t retrieved_vector_size=0;
+
+        myVectorFloats vec_f(i_vec_size);                                           // Load vector randomly with specified size.
+        std::string json_str = vec_f.serialize__Vector_to__JSON();                  // Now, use this vector to create a JSON serialization memory stream.
+#ifdef SAVE_FLOATS_TO_TEST_FILE                                                     // WARNING: Precompiler variable created through root CMakeLists file.
```

Comment:
does not make sense to use definition variables. Just include this part always or never.